experimental/devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,259 @@
+//
+// https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMP180-DS000-121.pdf
+//
+// The font the official datasheet on page 15 is unreadable, a copy with
+// readable text can be found here:
+//
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+//
+// Notes on the datasheet
+//
+// The results of the calculations in the algorithm on page 15 are partly
+// wrong. It looks like the original authors used non-integer calculations and
+// some nubers were rounded. Take the results of the calculations with a grain
+// of salt.
+package bmp180
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/devices"
+)
+
+// Oversampling affects how much time is taken to measure pressure.
+type Oversampling uint8
+
+// Possible oversampling values.
+const (
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+)
+
+// Dev is a handle to a bmp180.
+type Dev struct {
+	dev *mmr.Dev8
+	cal calibration
+	os  Oversampling
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("BMP180{%s}", d.dev.Conn)
+}
+
+// bit offsets in regCtrlMeas
+const (
+	ctrlMeasurementControlShift = 0
+	ctrlStartConversionShift    = 5
+	ctrlOversamplingShift       = 6
+)
+
+const (
+	cmdStartTempConv     uint8 = (1 << ctrlStartConversionShift) | (0x0E << ctrlMeasurementControlShift)
+	cmdStartPressureConv uint8 = (1 << ctrlStartConversionShift) | (0x14 << ctrlMeasurementControlShift)
+)
+
+const (
+	tempConvTime = 4500 * time.Microsecond // maximum conversion time for temperature
+)
+
+// maximum conversion time for pressure
+var pressureConvTime = [...]time.Duration{
+	4500 * time.Microsecond,
+	7500 * time.Microsecond,
+	13500 * time.Microsecond,
+	25500 * time.Microsecond,
+}
+
+// Sense returns measurements as °C and kPa.
+func (d *Dev) Sense(env *devices.Environment) error {
+	// start conversion for temperature
+	if err := d.dev.WriteUint8(regCtrlMeas, cmdStartTempConv); err != nil {
+		return err
+	}
+
+	time.Sleep(tempConvTime)
+
+	// read value
+	ut, err := d.dev.ReadUint16(regOutMSB)
+	if err != nil {
+		return err
+	}
+
+	temp := d.cal.compensateTemp(ut)
+
+	// start conversion for pressure
+	cmd := cmdStartPressureConv
+	cmd |= uint8(d.os) << ctrlOversamplingShift
+
+	if err := d.dev.WriteUint8(regCtrlMeas, cmd); err != nil {
+		return err
+	}
+
+	time.Sleep(pressureConvTime[d.os])
+
+	// read value
+	var pressureBuf [3]byte
+	if err := d.dev.ReadStruct(regOutMSB, pressureBuf[:]); err != nil {
+		return err
+	}
+
+	up := (int32(pressureBuf[0])<<16 + int32(pressureBuf[1])<<8 | int32(pressureBuf[2])) >> (8 - d.os)
+
+	pressure := d.cal.compensatePressure(up, int32(ut), d.os)
+
+	env.Temperature = devices.Celsius(temp * 100)
+	env.Pressure = devices.KPascal(pressure)
+	return nil
+}
+
+// Halt is a noop for the BMP180.
+func (d *Dev) Halt() error {
+	return nil
+}
+
+const (
```

Comment:
group all consts groups without functions in between.

but: I generally prefer public symbols to be first, the. private .
then each group has const, var, the code. it's not an hard requirement